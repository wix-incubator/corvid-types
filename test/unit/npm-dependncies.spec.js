const npmDependencies = require("../../src/dynamicTypes/npmDependencies");

describe("npm dependencies", () => {
  describe("should load types for npm modules", () => {
    it("get default type when request raw", () => {
      const rawNpmType = npmDependencies.getRaw("react");
      expect(rawNpmType).toMatchSnapshot();
    });

    it("get default type when request files and the namespaces is not @wix", async () => {
      const npmDtsFiles = await npmDependencies.getFiles({
        react: "16.8.6",
        "react-dom": "16.8.6"
      });
      expect(npmDtsFiles).toMatchSnapshot();
    });

    it("get proper type for @wix scoped npm module", async () => {
      global.fetch = jest.fn(() =>
        Promise.resolve({
          ok: true,
          text: () => "type SababaType = {}"
        })
      );
      const npmDtsFiles = await npmDependencies.getFiles({
        "@wix/react-velo": "1.0.0"
      });
      expect(npmDtsFiles).toMatchSnapshot();
      expect(npmDtsFiles[0].content).toBe("type SababaType = {}");
    });

    it("get default type for @wix scoped npm module if request failed", async () => {
      global.fetch = jest.fn(() =>
        Promise.resolve({
          ok: false
        })
      );
      const npmDtsFiles = await npmDependencies.getFiles({
        "@wix/react-velo": "1.0.0"
      });
      expect(npmDtsFiles).toMatchSnapshot();
      expect(npmDtsFiles[0].content).toContain(
        "declare module '@wix/react-velo/*'"
      );
    });

    it("get default type for @wix scoped npm module if fetch throws", async () => {
      global.fetch = jest.fn(() => Promise.reject(new Error("Fetch error")));
      const npmDtsFiles = await npmDependencies.getFiles({
        "@wix/react-velo": "1.0.0"
      });
      expect(npmDtsFiles).toMatchSnapshot();
      expect(npmDtsFiles[0].content).toContain(
        "declare module '@wix/react-velo/*'"
      );
    });
  });
});
