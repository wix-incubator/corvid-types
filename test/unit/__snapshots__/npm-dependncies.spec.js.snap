// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`npm dependencies should load types for npm modules get default type for @wix scoped npm module if fetch throws 1`] = `
Array [
  Object {
    "content": "
    declare module '@wix/react-velo';
    declare module '@wix/react-velo/*';
  ",
    "path": "/dependencies/@wix/react-velo.1.0.0.d.ts",
  },
]
`;

exports[`npm dependencies should load types for npm modules get default type for @wix scoped npm module if request failed 1`] = `
Array [
  Object {
    "content": "
    declare module '@wix/react-velo';
    declare module '@wix/react-velo/*';
  ",
    "path": "/dependencies/@wix/react-velo.1.0.0.d.ts",
  },
]
`;

exports[`npm dependencies should load types for npm modules get default type when request files and the namespaces is not @wix 1`] = `
Array [
  Object {
    "content": "
    declare module 'react';
    declare module 'react/*';
  ",
    "path": "/dependencies/react.16.8.6.d.ts",
  },
  Object {
    "content": "
    declare module 'react-dom';
    declare module 'react-dom/*';
  ",
    "path": "/dependencies/react-dom.16.8.6.d.ts",
  },
]
`;

exports[`npm dependencies should load types for npm modules get default type when request raw 1`] = `
"
    declare module 'react';
    declare module 'react/*';
  "
`;

exports[`npm dependencies should load types for npm modules get proper type for @wix scoped npm module 1`] = `
Array [
  Object {
    "content": "type SababaType = {}",
    "path": "/dependencies/@wix/react-velo.1.0.0.d.ts",
  },
]
`;
